From 8843a016dbf9edd7cae6051137ade5429e663edb Mon Sep 17 00:00:00 2001
From: Jacob Chen <jacob-chen@iotwrt.com>
Date: Wed, 28 Jun 2017 15:57:11 +0800
Subject: [PATCH 1/6] DEBIAN-ONLY: rockchipmpp: adaptations to 1.10

debian still use gstreamer 1.10
it should be compatible with debian

Signed-off-by: Jacob Chen <jacob-chen@iotwrt.com>
---
 configure.ac                      | 31 +++++++++++++-----------
 gst-libs/gst/gst-compat-private.h | 50 +++++++++++++++++++++++++++++++++++++++
 gst/rockchipmpp/gstmppvideodec.c  |  1 +
 3 files changed, 68 insertions(+), 14 deletions(-)
 create mode 100644 gst-libs/gst/gst-compat-private.h

diff --git a/configure.ac b/configure.ac
index 85459b4..6d9bfaf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5,7 +5,7 @@ dnl please read gstreamer/docs/random/autotools before changing this file
 dnl initialize autoconf
 dnl releases only do -Wall, git and prerelease does -Werror too
 dnl use a three digit version number for releases, and four for git/pre
-AC_INIT([GStreamer Plug-ins], [1.12.0], [https://github.com/rockchip-linux/gstreamer-rockchip/issues], [gstreamer-rockchip])
+AC_INIT([GStreamer Plug-ins], [1.10.0], [https://github.com/rockchip-linux/gstreamer-rockchip], [gstreamer-rockchip])
 
 AG_GST_INIT
 
@@ -46,7 +46,7 @@ AG_GST_LIBTOOL_PREPARE
 AS_LIBTOOL(GST, 801, 0, 801)
 
 dnl *** required versions of GStreamer stuff ***
-GST_REQ=1.12.0
+GST_REQ=1.10.0
 GSTPB_REQ=1.10.0
 
 dnl *** autotools stuff ****
@@ -57,6 +57,9 @@ AS_AUTOTOOLS_ALTERNATE
 dnl Add parameters for aclocal
 AC_SUBST(ACLOCAL_AMFLAGS, "-I m4 -I common/m4")
 
+dnl PKG_CONFIG_SYSROOT_DIR is a valid environment variable
+m4_pattern_allow(PKG_CONFIG_SYSROOT_DIR)
+
 dnl set up gettext
 dnl the version check needs to stay here because autopoint greps for it
 AM_GNU_GETTEXT_VERSION([0.17])
@@ -158,7 +161,7 @@ AC_CACHE_CHECK([for isinf], ac_cv_have_isinf,
     AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <math.h>]], [[float f = 0.0; int i=isinf(f)]])],[ac_cv_have_isinf="yes"],[ac_cv_have_isinf="no"]))
 if test "$ac_cv_have_isinf" = "yes"
 then
-    AC_DEFINE(HAVE_ISINF, 1, [Define to 1 if you have the 'isinf' function.])
+    AC_DEFINE(HAVE_ISINF, 1, [Define to 1 if you have the `isinf' function.])
 fi
 
 dnl Check for mmap (needed by electricfence plugin)
@@ -173,6 +176,7 @@ AC_CHECK_TYPE([struct ip_mreqn], [
 
 dnl *** checks for dependency libraries ***
 
+
 dnl GLib
 GLIB_REQ=2.32.0
 AG_GST_GLIB_CHECK([$GLIB_REQ])
@@ -276,9 +280,7 @@ AM_CONDITIONAL(GST_PLUGIN_BUILD_STATIC, test "x$enable_static_plugins" = "xyes")
 # set by AG_GST_PARSE_SUBSYSTEM_DISABLES above
 dnl make sure it doesn't complain about unused variables if debugging is disabled
 NO_WARNINGS=""
-if test "x${GST_DISABLE_GST_DEBUG}" = "xyes"; then
-  NO_WARNINGS="-Wno-unused"
-fi
+AG_GST_CHECK_GST_DEBUG_DISABLED([NO_WARNINGS="-Wno-unused"], [NO_WARNINGS=""])
 
 dnl define an ERROR_CFLAGS Makefile variable
 dnl -Wundef: too many broken headers
@@ -314,6 +316,7 @@ dnl these are all the gst plug-ins, compilable without additional libs
 dnl of EXT plugins
 AM_CONDITIONAL(USE_ROCKCHIPMPP, false)
 AM_CONDITIONAL(USE_VPUDEC, false)
+AM_CONDITIONAL(USE_KMS, false)
 AM_CONDITIONAL(USE_RKXIMAGE, false)
 
 dnl *** finalize CFLAGS, LDFLAGS, LIBS
@@ -401,21 +404,21 @@ AG_GST_CHECK_FEATURE(RKXIMAGE, [drm/kms and x11 libraries], rkximage, [
 
 dnl keep this alphabetic per directory, please
 AC_CONFIG_FILES(
-common/m4/Makefile
-common/Makefile
-m4/Makefile
 Makefile
-gst/Makefile
-gst/rkximage/Makefile
-gst/rockchipmpp/Makefile
-gst/vpudec/Makefile
 gst-libs/Makefile
 gst-libs/gst/Makefile
 gst-libs/gst/vpudec/Makefile
+gst/Makefile
+gst/vpudec/Makefile
+gst/rockchipmpp/Makefile
+gst/rkximage/Makefile
+common/Makefile
+common/m4/Makefile
+m4/Makefile
 tests/Makefile
 tests/examples/Makefile
-tests/examples/camera/Makefile
 tests/examples/player/Makefile
+tests/examples/camera/Makefile
 )
 
 AC_OUTPUT
diff --git a/gst-libs/gst/gst-compat-private.h b/gst-libs/gst/gst-compat-private.h
new file mode 100644
index 0000000..bc36f2c
--- /dev/null
+++ b/gst-libs/gst/gst-compat-private.h
@@ -0,0 +1,50 @@
+#ifndef __GST_COMPAT_PRIVATE_H__
+#define __GST_COMPAT_PRIVATE_H__
+
+#include <glib.h>
+
+G_BEGIN_DECLS
+
+/* copies */
+
+#if !GST_CHECK_VERSION(1,12,0)
+ /**
+ * gst_pad_get_task_state:
+ * @pad: the #GstPad to get task state from
+ *
+ * Get @pad task state. If no task is currently
+ * set, GST_TASK_STOPPED is returned.
+ *
+ * Returns: The current state of @pad's task.
+ */
+static inline GstTaskState
+gst_pad_get_task_state (GstPad * pad)
+{
+  GstTask *task;
+  GstTaskState res;
+
+  g_return_val_if_fail (GST_IS_PAD (pad), GST_TASK_STOPPED);
+
+  GST_OBJECT_LOCK (pad);
+  task = GST_PAD_TASK (pad);
+  if (task == NULL)
+    goto no_task;
+  res = gst_task_get_state (task);
+  GST_OBJECT_UNLOCK (pad);
+
+  return res;
+
+no_task:
+  {
+    GST_DEBUG_OBJECT (pad, "pad has no task");
+    GST_OBJECT_UNLOCK (pad);
+    return GST_TASK_STOPPED;
+  }
+}
+#endif
+
+/* adaptations */
+
+G_END_DECLS
+
+#endif
diff --git a/gst/rockchipmpp/gstmppvideodec.c b/gst/rockchipmpp/gstmppvideodec.c
index 432b54a..718a9fe 100644
--- a/gst/rockchipmpp/gstmppvideodec.c
+++ b/gst/rockchipmpp/gstmppvideodec.c
@@ -24,6 +24,7 @@
 #endif
 
 #include <gst/gst.h>
+#include <gst/gst-compat-private.h>
 
 #include "gstmppdecbufferpool.h"
 #include "gstmppvideodec.h"
-- 
2.15.1

